backend/test/app_configuration/policy_helper/test_notebook.py

```diff
@@ -0,0 +1,122 @@
+#
+#   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+#   Licensed under the Apache License, Version 2.0 (the "License").
+#   You may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+import json
+from unittest import mock
+
+from ml_space_lambda.data_access_objects.app_configuration import ServiceInstanceTypes
+from ml_space_lambda.enums import ResourceType, ServiceType
+from ml_space_lambda.utils import mlspace_config
+
+TEST_ENV_CONFIG = {
+    "AWS_DEFAULT_REGION": "us-east-1",
+    "JOB_INSTANCE_CONSTRAINT_POLICY_ARN": "arn:aws:iam::policy/job-instance-constraint",
+    "ENDPOINT_CONFIG_INSTANCE_CONSTRAINT_POLICY_ARN": "arn:aws:iam::policy/job-instance-constraint",
+}
+
+mock_context = mock.Mock()
+mock_context.invoked_function_arn.split.return_value = "arn:aws:lambda:us-east-1:123456789010:function/some-lambda".split(":")
+
+with mock.patch.dict("os.environ", TEST_ENV_CONFIG, clear=True):
+    from ml_space_lambda.app_configuration.lambda_functions import update_instance_constraint_policies
+    from ml_space_lambda.app_configuration.policy_helper.notebook import (
+        create_instance_constraint_policy_version,
+        create_instance_constraint_statement,
+        create_sagemaker_resource_arn,
+    )
+
+
+@mock.patch.dict("os.environ", TEST_ENV_CONFIG, clear=True)
+@mock.patch("ml_space_lambda.app_configuration.policy_helper.notebook.iam")
+def test_update_instance_constraint_policies(iam):
+    # Clear out previously cached env variables
+    mlspace_config.env_variables = {}
+
+    new_configuration = ServiceInstanceTypes.from_dict(
+        {
+            ServiceType.NOTEBOOK.value: ["ml.t3.medium"],
+            ServiceType.ENDPOINT.value: ["ml.t3.large"],
+            ServiceType.TRAINING_JOB.value: ["ml.t3.xlarge"],
+            ServiceType.TRANSFORM_JOB.value: ["ml.t3.kindabig"],
+        }
+    )
+
+    update_instance_constraint_policies(new_configuration, mock_context)
+    iam.create_policy_version.assert_has_calls(
```

Comment:
If this is just intended to test the update function in isolation, then the mock should probably be of the `create_instance_constraint_policy_version` function. If this is intended to test overall success it seems that it would make sense ensuring the proper statements were used for the call instead of `mock.ANY`.

Won't block, because this is mostly covered already.